### ISSUE TITLE
Add custom validator for `JupyterScatter.points` and `JupyterScatter.selection`

### ISSUE THREAD

**Issue by manzt** [flekschas/jupyter-scatter#14]
After #11, we use `traittypes.Array` to annotate `JupyterScatter.points` and `JupyterScatter.selection`. 

However, `traittypes.Array` is valid for _any_ numpy array and the `shape`/`dtype` for `points` and `selection` are more strict. We should add [custom validators](https://github.com/jupyter-widgets/traittypes#example-usage-with-a-custom-validator) for these attributes, which will raise an exception if an array of the wrong shape/dtype is assigned on the python side. 

This isn't urgent, since `jscatter.Scatter` for the most part ensures the data are correct, but this would probably help with tracking down weird bugs of where dtypes aren't what are expected.

**Comment by flekschas** [comment]
Closing this for now as it's not a priority
